nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -178,4 +159,12 @@ public int getScriptHeaderFlags() {
 	public DataType getHeaderDataType() {
 		return this.scriptHeader.toDataType();
 	}
+
+	public DataType getBlockHeaderDataType() {
+		return this.blockHeader.toDataType();
+	}
+
+	public InputStreamByteProvider getInputStreamByteProvider() {
```

Comment:
This function should be something like:

```
pubic InputStream getBlockInputStream() {
  return this.nsisDecompressionProvider.getDecompressedStream();
}
```

When you are parsing the NSIS file when you figure out the compression method, if you save the instance of the `NsisDecompressionProvider` you create you can call this function anytime you need the `InputStream` of the deccompressed bytes. The caller won't need to know the offset the bytes start at or their size, the caller can simple read from the `InputStream` until the bytes end.

Reply:
Updated